stm32 devices with USB-C PD cannot use CC1 and CC2 pins by default

FRASTM opened the pull request:
It disables USB Type-C dead battery pull-down behavior on UCPDx_CC1 and UCPDx_CC2 pins
of the stm32G4 and stm32L5  and stm32G0 soc series which have the USB type C Power Delivery feature. 

After exiting reset, the USB Type-C “dead battery” behavior is enabled, which may have a
pulldown effect on CC1 and CC2 pins. It is recommended to disable it in all cases, either to
stop this pull-down or to hand over control to the UCPD (which should therefore be initialized
before doing the disable).

For stm32l5 and stm32g4, the UCPD DBDIS is controlled by the UCPD1_DBDIS bit of the PWR CR3 register.
For stm32g0, the UCPD DBDIS are controlled by the strobe bits of the SYSCFG CFGR1 register

Fix  #33253  

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
soc: arm: stm32g4 with USB-C PD cannot use CC1 and CC2 pins by default

With this patch, the UCPD1 _CC1 and _CC2 pins
are disabling the USB Type-C and Power Delivery Dead Battery.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
soc: arm: stm32g0 with USB-C PD cannot use CC1 and CC2 pins by default

With this patch, the UCPD1 _CC1 and _CC2 pins
are disabling the USB Type-C and Power Delivery Dead Battery.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

erwango commented on soc/arm/st_stm32/stm32g4/soc.c:
```diff
@@ -45,6 +49,11 @@ static int stm32g4_init(const struct device *arg)
 
+#if defined(PWR_CR3_UCPD_DBDIS)
+	/* Disable USB Type-C dead battery pull-down behavior */
+	__HAL_RCC_PWR_CLK_ENABLE();
```
Please use LL API

FRASTM replied:
LL api is LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR) from the stm32_ll_bus.h

erwango commented on soc/arm/st_stm32/stm32l5/soc.c:
```diff
@@ -47,6 +47,11 @@ static int stm32l5_init(const struct device *arg)
 
+#if defined(PWR_CR3_UCPD_DBDIS)
+	/* Disable USB Type-C dead battery pull-down behavior */
+	__HAL_RCC_PWR_CLK_ENABLE();
```
LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR); is called just before.

FRASTM replied:
removed

FRASTM commented:
with LL API

erwango commented:
^^ @str4t0m @molchec

FRASTM commented:
including SYSCFG clock control

erwango commented on soc/arm/st_stm32/stm32g4/soc.c:
```diff
@@ -15,6 +15,11 @@
 #include <arch/cpu.h>
 #include <arch/arm/aarch32/cortex_m/cmsis.h>
 
+#if defined(PWR_CR3_UCPD_DBDIS)
```
`PWR_CR3_UCPD_DBDIS` is defined for all G4 variants.
In that regard, this condition is superfluous.

erwango commented on soc/arm/st_stm32/stm32g4/soc.c:
```diff
@@ -45,6 +50,11 @@ static int stm32g4_init(const struct device *arg)
 	/* allow reflashing board */
 	LL_DBGMCU_EnableDBGSleepMode();
 
+#if defined(PWR_CR3_UCPD_DBDIS)
```
ditto

FRASTM replied:
yes, 
however, the `stm32g4xx_ll_pwr.h` has this flag  `#if defined(PWR_CR3_UCPD_DBDIS)`

erwango commented on soc/arm/st_stm32/stm32l5/soc.c:
```diff
@@ -47,6 +47,10 @@ static int stm32l5_init(const struct device *arg)
 	LL_APB1_GRP1_EnableClock(LL_APB1_GRP1_PERIPH_PWR);
 	LL_PWR_SetRegulVoltageScaling(LL_PWR_REGU_VOLTAGE_SCALE0);
 
+#if defined(PWR_CR3_UCPD_DBDIS)
```
ditto

FRASTM replied:
removed

erwango replied:
still there...

FRASTM committed:
soc: arm: stm32l5 with USB-C PD cannot use CC1 and CC2 pins by default

With this patch, the UCPD1 _CC1 and _CC2 pins
are disabling the USB Type-C and Power Delivery Dead Battery.

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
removing the PWR_CR3_UCPD_DBDIS  flag for the stm32L5 serie

martinjaeger reviewed (approved):
Looks good. Thanks for implementing the fix.
